### PR TITLE
Add support for enforcing client tenantid when using host resolution …

### PR DESCRIPTION
**Pull request by Torgeir-Hansen**

### CODE REVIEW

File: Source/Configuration/TenantConfig.cs [hunk 1]
```diff
@@ -19,10 +19,12 @@ public record TenantConfig
...
-    /// <remarks>
-    /// If the tenant specified is the actual tenant, it is not impersonating that tenant.
-    /// </remarks>
     public string[] SourceIdentifiers { get; set; } = Array.Empty<string>();
-}
+
+    /// <summary>
+    /// Gets or sets a collection of entra id tenant identifiers allowed to log in with this middleware tenant.
+    /// </summary>
+    public string[] EntraIdTenants { get; set; } = Array.Empty<string>();
```
Comment (smithmx) [inline Source/Configuration/TenantConfig.cs:29]: I think the use of Tenants here is confusing.  It refers to an EntraId tenant (basically an instance) but not our "tenant" that we are potentially impersonating.  I'd drop the use of "tenant" for Entra even though that's the term Microsoft use.  Maybe use something like source or instance to distinguish from our own tenant?

File: Source/RequestAugmenter.cs [hunk 2]
```diff
@@ -115,7 +115,7 @@ public async Task<IActionResult> Get()
         }
 
         // Finally check the entra id requirement.
-        return _roleAuthorizer.Handle(Request);
+        return _roleAuthorizer.Handle(Request, tenantId);
```
Comment (smithmx) [inline Source/RequestAugmenter.cs:118]: The comment is a bit confusing here.  It refers to the entra id but the method sig takes in the tenant id.

I'd change the comment to say something like "Confirm that the user has the correct role / permissions for this tenant".

**Review by smithmx — APPROVED** [review]
Good catch!  Should have been done at the time of writing.

I've some minor comments about the naming (just that I find it confusing to use both the EntraId tenant terminology and our usage of Tenant).  It's not critical though.  Feel free to go ahead with this.

Or if you want, I can make the changes around naming / comments to save you the typing.

Really liked the test btw.  Very clear.

File: Source/Configuration/TenantConfig.cs (hunk 1 above)
Comment (Torgeir-Hansen) [reply Source/Configuration/TenantConfig.cs:29]: Introducing another term for what is globally known as the EntraId Tenant Id would probably cause more confusion 😅

Implicitly, when referring to the apps own concept of tenant ids, only tenantid should be used.
When referring to external forms of tenant id, they should be explicit.

To add to the brainfuck, a entra id tenant id can also be used just as a source identifier 😆
This middleware really needs to be cleaned up at a later time, especially with regards to the configuration structure - as it is too easy to make mistakes without realizing it.

**Pull request merged**